### DISCUSSION TITLE
Would adding py.typed suffice to add mypy support ?

### DISCUSSION THREAD

**Discussion by JonathanPlasse** [geospace-code/pymap3d#73] [Ideas]
Hi,
Love pymap3d.

When running my code with mypy without `--ignore-missing-imports`, I get the error `Skipping analyzing "pymap3d": module is installed, but missing library stubs or py.typed marker`.
Would adding py.typed suffice to add mypy support as described [here](https://blog.whtsky.me/tech/2021/dont-forget-py.typed-for-your-typed-python-package/)?

**Comment by JonathanPlasse** [comment]
Never mind, it would cause problems as the current typing would cause problems.
Returning `tuple` instead of `tuple[float, float, float]` for example.

**Reply by scivision** [reply]
Is there a way that doesn't cause problems when the same function can return either `tuple[float, float, float]` or `tuple[numpy.ndarray, numpy.ndarray, numpy.ndarray]`

**Reply by JonathanPlasse** [reply]
Here is a solution

```python
try:
    from numpy.typing import NDArray
except ImportError:
    pass

@overload
def enu2aer(e: float, n: float, u: float, deg: bool = True) -> Tuple[float, float, float]:
    pass


@overload
def enu2aer(
    e: "NDArray", n: "NDArray", u: "NDArray", deg: bool = True
) -> Tuple["NDArray", "NDArray", "NDArray"]:
    pass
```

If numpy is installed, vscode hints would show,

```python
enu2aer(e: float, n: float, u: float, deg: bool = True) -> Tuple[float, float, float]

enu2aer(e: NDArray, n: NDArray, u: NDArray, deg: bool = True) -> Tuple[NDArray, NDArray, NDArray]
```

Otherwise,

```python
enu2aer(e: float, n: float, u: float, deg: bool = True) -> Tuple[float, float, float]

enu2aer(e: Any, n: Any, u: Any, deg: bool = True) -> Tuple[Any, Any, Any]
```

I am working on a PR to apply this to the rest of the code

**Comment by JonathanPlasse** [comment]
The pull request #76 resolves this problem.
